### PR TITLE
Features/autocreate profiles

### PR DESCRIPTION
**Pull request by vtymosh**
- creates for every device one predefined profile if no profiles yet exist
- use non-root user for Docker image
- move OeeAPI to its own file
- improve logging
- fix minor issues

### CODE REVIEW

File: event-based-simulators/main/event_based_simulators.py [hunk 1]
```diff
@@ -11,7 +11,6 @@ def current_timestamp(format = "%Y-%m-%dT%H:%M:%S.%f"):
 
 logging.basicConfig(format='%(asctime)s %(name)s:%(message)s', level=logging.INFO)
 log = logging.getLogger("sims")
-log.info(os.environ)
```
Comment (mvoigt-sag) [inline event-based-simulators/main/event_based_simulators.py]: what is the reason to not log the environ?

**Review by mvoigt-sag — APPROVED** [review]
I didnt do a testrun, but the code changes look good to me. question is: should we still provide a docker image that does _not_ create the profiles automatically?

File: event-based-simulators/main/event_based_simulators.py (hunk 1 above)
Comment (vtymosh) [reply event-based-simulators/main/event_based_simulators.py]: password.

**Comment by vtymosh** [comment]
- creates for every device one predefined profile if no profiles yet exist
 - use non-root user for Docker image
 - move OeeAPI to its own file
 - improve logging
 - fix minor issues

**Comment by thomaswinkler** [comment]
> I didnt do a testrun, but the code changes look good to me. question is: should we still provide a docker image that does _not_ create the profiles automatically?

It would be good if there was an option not to create / recreate the profiles automatically. Not sure what the best option would be, but my first thought has also been to have 2 docker images. 

Is there any way we can configure / pass parameters to the Microservice?

**Comment by vtymosh** [comment]
> 
> 
> > I didnt do a testrun, but the code changes look good to me. question is: should we still provide a docker image that does _not_ create the profiles automatically?
> 
> It would be good if there was an option not to create / recreate the profiles automatically. Not sure what the best option would be, but my first thought has also been to have 2 docker images.
> 
> Is there any way we can configure / pass parameters to the Microservice?

AFAIK, it's not really possible. Microservice gets just few predefined environment variables from Cumulocity. We need make it configuratble at build time and just provide few different docker images.

**Pull request merged**